agent/util-scripts/pbench-tool-meister-start

```diff
@@ -557,6 +562,40 @@ class ToolDataSink(BaseServer):
         if self.wait_for_pid(self.get_pid()):
             self.kill(status)
 
+    @staticmethod
+    def is_running(pid: int) -> bool:
+        """Is the given PID running?
+
+        See https://stackoverflow.com/questions/7653178/wait-until-a-certain-process-knowing-the-pid-end
+
+        Return True if a PID is running, else False if not.
+        """
+        try:
+            os.kill(pid, 0)
+        except ProcessLookupError:
+            return False
+        return True
+
+    @staticmethod
+    def wait_for_pid(pid: int) -> bool:
+        """wait for a process to actually stop running, but eventually give up
+        to avoid a hang."""
+        running = True
+        timeout = time.time() + _TDS_STARTUP_TIMEOUT
+        while running:
+            running = __class__.is_running(pid)
+            if running:
+                if time.time() >= timeout:
+                    break
+                time.sleep(0.1)
+        return running
```

Comment:
As I [mentioned elsewhere](https://github.com/distributed-system-analysis/pbench/pull/2734#discussion_r847500473), this code seems a little cumbersome.  If this PR is an opportunity to change it, I would pull for this instead:
```
        timeout = time.time() + _TDS_STARTUP_TIMEOUT
        while time.time() < timeout:
            if not __class__.is_running(pid):
                return False
            time.sleep(0.1)
        return True
```

Reply:
I will check if this change aligns with my PR I will consider refactoring it, else we might have to add a follow up PR :)